tests/app/auth/session-selection/session-selection.component.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { ErrorHandlingService } from '../../../../src/app/core/services/error-handling.service';
+import { GithubService } from '../../../../src/app/core/services/github.service';
+import { LoggingService } from '../../../../src/app/core/services/logging.service';
+import { PhaseService } from '../../../../src/app/core/services/phase.service';
+import { SharedModule } from '../../../../src/app/shared/shared.module';
+
+@Component({
+  selector: 'app-profiles',
+  template: ''
+})
+class ProfilesStubComponent implements Partial<ProfilesComponent> {
+  @Output() selectedProfileEmitter: EventEmitter<Profile> = new EventEmitter<Profile>();
+  @Input() urlEncodedSessionName: string;
+
+  selectProfile(profile: Profile): void {
+    this.selectedProfileEmitter.emit(profile);
+  }
+}
+
+describe('SessionSelectionComponent (unit tests)', () => {
+  let fixture: ComponentFixture<SessionSelectionComponent>;
+  let component: SessionSelectionComponent;
+  let debugElement: DebugElement;
+  let nativeElement: HTMLElement;
+  let profilesDebugEl: DebugElement;
+  let profilesComponent: ProfilesComponent;
+
+  const logger = jasmine.createSpyObj('LoggingService', ['info']);
+  const githubService = jasmine.createSpyObj('GithubService', ['storeOrganizationDetails']);
+  const phaseService = jasmine.createSpyObj('PhaseService', ['storeSessionData']);
+  const authService = jasmine.createSpyObj('AuthService', ['startOAuthProcess', 'changeAuthState']);
+  const errorHandlingService = jasmine.createSpyObj('ErrorHandlingService', ['handleError']);
+
+  const testProfile: Profile = {
+    profileName: 'testProfile',
+    repoName: 'testOrg/pe'
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [SharedModule, BrowserAnimationsModule],
+      declarations: [SessionSelectionComponent, ProfilesStubComponent],
+      providers: [
+        { provide: LoggingService, useValue: logger },
+        { provide: GithubService, useValue: githubService },
+        { provide: PhaseService, useValue: phaseService },
+        { provide: AuthService, useValue: authService },
+        { provide: ErrorHandlingService, useValue: errorHandlingService }
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SessionSelectionComponent);
+    fixture.detectChanges(); // onInit()
+    component = fixture.componentInstance;
+    debugElement = fixture.debugElement;
+    nativeElement = debugElement.nativeElement;
+    profilesDebugEl = debugElement.query(By.directive(ProfilesStubComponent));
+    profilesComponent = profilesDebugEl.componentInstance;
+  });
+
+  it('renders without errors', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('renders the profiles component', () => {
+    expect(profilesComponent).toBeTruthy();
+  });
+
+  it('should update the session input correctly when profile is selected', () => {
+    profilesComponent.selectProfile(testProfile);
```

Comment:
I think we can test the `ProfilesComponent` separately, and instead directly emit a Profile event here to trigger the change in profile.